lib/elixir/lib/macro.ex

```diff
@@ -317,6 +317,21 @@ defmodule Macro do
   end
 
   @doc """
+  Generates AST nodes for a given number of required argument variables using
+  `Macro.var/2`.
+
+  ## Examples
+
+      iex> Macro.generate_arguments(2, __MODULE__)
+      [{:var1, [], __MODULE__}, {:var2, [], __MODULE__}]
+
+  """
+  def generate_arguments(0, _), do: []
+  def generate_arguments(number_of_args, module) do
```

Comment:
Let's name second argument as `context`, we also probably want `is_integer/1`, `> 0`, and `is_atom/1` guards here.